libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1317,10 +1350,34 @@ void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Ty
 		m_context << Instruction::SUB;
 		break;
 	case Token::Mul:
-		m_context << Instruction::MUL;
+		if (c_isFractional)
+		{
+			//multiply, then shift right...this is your fraction.
```

Comment:
Can you explain what you do here, because I fear it is not correct - you are discarding overflow:

Let me try to explain that for decimal fixed point numbers. Assume our machine numbers are exactly 4 digits wide. We use two digits for integers, two digits for fractions and want to calculate:
```
22.22 * 04.00 = 88.88
```

The initial approach is to treat those numbers as integers and then divide afterwards:
```
2222 * 400 = 888800, divided by 10000 is 8888, which is 88.88
```
The problem is that the intemediate result, `888800` does not fit our machine words and it would be truncated.

This means that, as soon as the type we are using takes more than 128 bits (it actually depends on the number of fractional bits, too - we should optimize all cases), we have to multiply in a way that is a bit more complicated.

A good explanation can be found here: https://stackoverflow.com/questions/15123638/fixed-point-multiplication-without-64-bit-temporary